src/TraceEvent/TraceEvent.cs

```diff
@@ -2943,6 +2965,19 @@ internal protected void Dispatch(TraceEvent anEvent)
             try
             {
 #endif
+                // check if there is a registered dispatch hook
+                if (DispatchHook != null)
+                {
+                    // check if we should dispatch this event, the first to indicate
+                    // false we will return without dispatching
+                    var hook = DispatchHook;
+                    while (hook != null && hook.OnDispatch != null)
```

Comment:
:bulb: Ensure that `hook.OnDispatch` can never be null and remove the unnecessary null check